pkg/apis/submariner.io/v1/string.go

```diff
@@ -0,0 +1,17 @@
+package v1
+
+import (
+	"encoding/json"
+
+	"k8s.io/klog"
+)
+
+// String method returns a string representation, for human readability.
+func (e *Endpoint) String() string {
+
+	bytes, err := json.Marshal(e.Spec)
```

Comment:
I think we should marshal the whole object - it's useful to see at least some of the Object metadata too.

Reply:
```
 {
            "apiVersion": "submariner.io/v1",
            "kind": "Endpoint",
            "metadata": {
                "creationTimestamp": "2020-02-14T10:39:28Z",
                "generation": 1,
                "name": "cluster2-submariner-cable-cluster2-172-17-0-3",
                "namespace": "submariner-operator",
                "resourceVersion": "373557",
                "selfLink": "/apis/submariner.io/v1/namespaces/submariner-operator/endpoints/cluster2-submariner-cable-cluster2-172-17-0-3",
                "uid": "46945da8-4f16-11ea-af6e-0242ac110008"
            },
            "spec": {
                "backend": "ipsec",
                "cable_name": "submariner-cable-cluster2-172-17-0-3",
                "cluster_id": "cluster2",
                "hostname": "cluster2-worker",
                "nat_enabled": false,
                "private_ip": "172.17.0.3",
                "public_ip": "",
                "subnets": [
                    "100.95.0.0/16",
                    "10.245.0.0/16"
                ]
            }
        },
```

vs

```
 {
                "backend": "ipsec",
                "cable_name": "submariner-cable-cluster2-172-17-0-3",
                "cluster_id": "cluster2",
                "hostname": "cluster2-worker",
                "nat_enabled": false,
                "private_ip": "172.17.0.3",
                "public_ip": "",
                "subnets": [
                    "100.95.0.0/16",
                    "10.245.0.0/16"
                ]
        }
```

Reply:
True about the trace level, I guess the generation/resourceversion/timestamps could be useful.